target/linux/realtek/image/rtl930x_nand.mk

```diff
@@ -1 +1,26 @@
 # SPDX-License-Identifier: GPL-2.0-only
+
+include ./common.mk
+
+define Device/linksys_lgs328c
+  $(Device/uimage-rt-loader)
+  SOC := rtl9301
+  IMAGE_SIZE := 30720k
+  DEVICE_VENDOR := Linksys
+  DEVICE_MODEL := LGS328C
+  BELKIN_MODEL := BKS-RTL93xx
+  BELKIN_HEADER := 0x07600001
+  LINKSYS_MODEL := 60412040
+  PAGESIZE := 2048
+  BLOCKSIZE := 128k
+  UBINIZE_OPTS := -E 5
+  KERNEL := \
+	$$(KERNEL) | \
+	pad-to $$(BLOCKSIZE)
```

Comment:
Padding kernel is probably not needed (still doesn't hurt). What is needed is specifying KERNEL_SIZE, since kernel has dedicated partition with fixed size.

As side note, looks like Linksys uses backup partition in case first fails. Consider using that facility now, otherwise changing current approach to what Linksys implemented later, after first stable OpenWrt release, would make migrating a nightmare and probably need manual intervention.
The examples of using Linksys A/B scheme in ipx40xx, ipq806xx or mvebu.